sphinx_search/_static/js/rtd_sphinx_search.js

```diff
@@ -89,31 +117,41 @@ const get_section_html = (sectionData, page_link) => {
                 <span class="search__result__subheading"> \
                     <%= section_subheading %> \
                 </span> \
-                <p class="search__result__content"> \
-                    <%= section_content %> \
-                </p> \
+                <% for (var i = 0; i < section_content.length; ++i) { %> \
+                    <p class="search__result__content"> \
+                        <%= section_content[i] %> \
+                    </p> \
+                <% } %>\
             </div> \
         </a> \
         <br class="br-for-hits">';
 
     let section_subheading = sectionData._source.title;
-
-    if (
-        sectionData.highlight["sections.title"] !== undefined &&
-        sectionData.highlight["sections.title"].length >= 1
-    ) {
-        section_subheading = sectionData.highlight["sections.title"][0];
+    let highlight = sectionData.highlight;
+    if (getHighlightListData(highlight, "sections.title")) {
+        section_subheading = getHighlightListData(
+            highlight,
+            "sections.title"
+        )[0];
     }
 
-    let section_content =
-        sectionData._source.content.substring(0, 100) + " ...";
-
-    if (
-        sectionData.highlight["sections.content"] !== undefined &&
-        sectionData.highlight["sections.content"].length >= 1
-    ) {
-        section_content =
-            "... " + sectionData.highlight["sections.content"][0] + " ...";
+    let section_content = [
+        sectionData._source.content.substring(0, 100) + " ..."
```

Comment:
I'd like to see `100` be a constant defined at the top of the file, at least. We can keep that for now.

Reply:
Updated the PR.